### PR TITLE
Add keyless signing for all artifacts

### PR DESCRIPTION
**Pull request by idunbarh**
This PR contains three changes:

- This PR adds keyless signing of checksum.txt, artifacts, and packages using cosign
- Instructions were added on how to verify signatures
- build artifacts from the goreleaser are saved for 3 days

Closes: #34

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -61,12 +61,27 @@ builds:
...
 
 signs:
-  - id: cosign
-    signature: $artifact.sig
+  - id: checksum-keyless-signing
+    signature: "${artifact}-keyless.sig"
+    certificate: "${artifact}-keyless.pem"
     cmd: cosign
-    args: [sign-blob, --output-signature=$artifact.sig, --key=env://COSIGN_PRIVATE_KEY, $artifact]
+    args: [sign-blob, --output-signature, "${artifact}-keyless.sig", --output-certificate, "${artifact}-keyless.pem", $artifact]
```
Comment (jhoward-lm) [inline .goreleaser.yml:68]: Will this work?

```suggestion
    args: [sign-blob, --output-signature=$signature, --output-certificate=$certificate, $artifact]
```

File: .goreleaser.yml [hunk 2]
```diff
@@ -61,12 +61,27 @@ builds:
...
     artifacts: checksum
 
+  - id: binary-keyless-signing
+    signature: "${artifact}-keyless.sig"
+    certificate: "${artifact}-keyless.pem"
+    cmd: cosign
+    args: [sign-blob, --output-signature, "${artifact}-keyless.sig", --output-certificate, "${artifact}-keyless.pem", $artifact]
```
Comment (jhoward-lm) [inline .goreleaser.yml:75]: Will this work?

```suggestion
    args: [sign-blob, --output-signature=$signature, --output-certificate=$certificate, $artifact]
```

File: .goreleaser.yml [hunk 3]
```diff
@@ -61,12 +61,27 @@ builds:
...
+    artifacts: binary
+
+  - id: package-keyless-signing
+    signature: "${artifact}-keyless.sig"
+    certificate: "${artifact}-keyless.pem"
+    cmd: cosign
+    args: [sign-blob, --output-signature, "${artifact}-keyless.sig", --output-certificate, "${artifact}-keyless.pem", $artifact]
```
Comment (jhoward-lm) [inline .goreleaser.yml:82]: Will this work?

```suggestion
    args: [sign-blob, --output-signature=$signature, --output-certificate=$certificate, $artifact]
```

**Review by jhoward-lm — CHANGES_REQUESTED** [review]
Just one small suggestion in 3 places to leverage the `$certificate` and `$signature` variables in cosign args.

I don't have a strong preference about

`--output-signature=$signature`

versus

`--output-signature, $signature`

I do prefer to omit quotes in YAML documents wherever syntactically possible, so these suggested changes allow removal of the braces around the variable name, e.g. `"${signature}"` --> `$signature`

File: .github/workflows/goreleaser.yml [hunk 4]
```diff
@@ -47,7 +48,9 @@ jobs:
           cache: false
 
       - name: Install Cosign
-        uses: sigstore/cosign-installer@e1523de7571e31dbe865fd2e80c5c7c23ae71eb4 # v3.4.0
+        uses: sigstore/cosign-installer@59acb6260d9c0ba8f4a2f9d9b48431a222b68e20 # v3.5.0
+        with:
+          cosign-release: "v2.2.4" # optional
```
Comment (jhoward-lm) [inline .github/workflows/goreleaser.yml:53]: Mind removing the quotes and comment?

```suggestion
          cosign-release: v2.2.4
```

Unless this accepts SHA references;  if so, my suggestion would instead be:

```yaml
          cosign-release: fb651b4ddd8176bd81756fca2d988dd8611f514d # v2.2.4
```

**Comment by jhoward-lm** [comment]
Also feel free to remove the `commitlint` workflow that checks the PR title. Since we're not using `svu` to generate the next release version and instead cutting releases via manual tag push now, it's a false positive

**Comment by jhoward-lm** [comment]
> I was planning leaving the shell script that wraps `svu` incase we want to run it locally. Thoughts?

Works for me

File: .goreleaser.yml (hunk 2 above)
Comment (jhoward-lm) [reply .goreleaser.yml]: Maybe theirs was originally written before goreleaser added support for that variable?

**Review by jhoward-lm — APPROVED** [review]

**Review by ashearin — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -72,3 +72,28 @@ TBD
...
+
+Bomctl releases can be [found here](https://github.com/bomctl/bomctl/releases) and are signed
+using keyless signing with cosign.
+
+For each artifact there are two additional files:
+
+- `${artifact}-keyless.sig` - signature
+- `${artifact}-keyless.pem` - certificate
+
+You can then verify this artifact with cosign the signature and the certificate.
```
Comment (mfrystacky) [inline README.md:86]: You can then verify this artifact with cosign **using** the signature and the certificate.

File: README.md [hunk 6]
```diff
@@ -72,3 +72,28 @@ TBD
...
+
+``` shell
+cosign verify-blob --certificate ${artifact}-keyless.pem --signature ${artifact}-keyless.sig --certificate-oidc-issuer https://token.actions.githubusercontent.com --certificate-identity-regexp 'https://github\.com/bomctl/bomctl/\.github/.+'  ${artifact}
+Verified OK
```
Comment (mfrystacky) [inline README.md:90]: Is this part of the command or the output?

**Review by mfrystacky — APPROVED** [review]
Had a question and a small fix

**Pull request merged**